routes/projects/projectRouter.js

```diff
@@ -0,0 +1,77 @@
+const express = require("express");
+const db = require("../../data/helpers/projectModel");
+
+const router = express.Router();
+
+//get all projects
+router.get("/", async (req, res) => {
+  try {
+    const allProjects = await db.get();
+    res.json({ message: allProjects });
+  } catch (err) {
+    res.status(500).json({ message: "internal server erros" });
+  }
+});
+
+//get project by ID
+router.get("/:id", validateID, async (req, res) => {
+  const singleProject = await db.get(req.params.id);
+  res.json({ message: singleProject });
+});
+
+//add project
+router.post("/", async (req, res) => {
+  console.log(req.body);
+  try {
+    const newProject = await db.insert(req.body);
+    res.json({ message: newProject });
+  } catch (err) {
+    res.status(500).json({ message: "internal server error" });
+  }
+});
+
+//update action
+router.put("/:id", validateID, async (req, res) => {
+  try {
+    const updatedProject = await db.update(req.params.id, req.body);
+    res.json({ message: updatedProject });
+  } catch (err) {
+    res.status(500).json({ message: "Internal Server Error" });
+  }
+});
+
+//remove project
+router.delete("/:id", validateID, async (req, res) => {
+  try {
+    const deletedProject = await db.remove(req.params.id);
+    res.json({ message: deletedProject });
+  } catch (err) {
+    res.status(500).json({ message: "Internal Server Error" });
+  }
+});
+
+//getProjectActions
+router.get("/:id/actions", validateID, async (req, res) => {
+  const { id } = req.params;
```

Comment:
Good job here. Functionally if we made a react app we would want to click on a single project, go to it's page and see all of the details. How could you set up one endpoint that returns all of the actions for a project without making multiple requests on the front end?